### PR TITLE
Set container

### PR DESCRIPTION
**Pull request by Ypostolov**

### CODE REVIEW

File: week_2/bus_stops_3/bus_stops_3.cpp [hunk 1]
```diff
@@ -0,0 +1,34 @@
+#include <iostream>
+#include <map>
+#include <set>
+#include <string>
+
+int main() {
+  int Q;
```
Comment (ZhdanovP) [inline week_2/bus_stops_3/bus_stops_3.cpp:7]: @Ypostolov what is the data will be stored in Q ? Why this variable has different style of naming?

File: week_2/bus_stops_3/bus_stops_3.cpp [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  std::cin >> Q;
+  std::string stop;
+  int stops_count;
+  int k = 1;
```
Comment (ZhdanovP) [inline week_2/bus_stops_3/bus_stops_3.cpp:11]: @Ypostolov which info will be stored in var k? For what purpose we need it?

File: week_2/bus_stops_3/bus_stops_3.cpp [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  std::set<std::string> stops;
+  std::map<std::set<std::string>, int> route;
```
Comment (ZhdanovP) [inline week_2/bus_stops_3/bus_stops_3.cpp:13]: @Ypostolov Why the route contains set as a key? Do we really need such complex structure?

File: week_2/bus_stops_3/bus_stops_3.cpp [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+  for (int it = 0; it < Q; ++it) {
+    std::cin >> stops_count;
+
+    for (int i = 0; i < stops_count; ++i) {
+      std::cin >> stop;
+      stops.insert(stop);
+    }
+
+    if (route.count(stops) != 0) {
```
Comment (ZhdanovP) [inline week_2/bus_stops_3/bus_stops_3.cpp:23]: @Ypostolov please, give example of `route`  and `stops` when `route.count(stops)` will be equal to 3

File: week_2/bus_stops_3/bus_stops_3.cpp [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+      std::cout << "Already exists for " << route[stops] << std::endl;
+    } else {
+      route[stops] = k;
+      std::cout << "New bus " << route[stops] << std::endl;
+      ++k;
+    }
+    stops.clear();
```
Comment (ZhdanovP) [inline week_2/bus_stops_3/bus_stops_3.cpp:30]: @Ypostolov Why should we clear stops ? What if remove this line?

File: week_2/bus_stops_3/bus_stops_3.cpp (hunk 1 above)
Comment (Ypostolov) [reply week_2/bus_stops_3/bus_stops_3.cpp:7]: In the "Q" will be stored number of requests, style is according task description `Сначала вводится количество запросов Q`

File: week_2/bus_stops_3/bus_stops_3.cpp (hunk 2 above)
Comment (Ypostolov) [reply week_2/bus_stops_3/bus_stops_3.cpp:11]: in this variable we store new numbers for buses routs, we assign new route numbers according to the value of this variable

File: week_2/bus_stops_3/bus_stops_3.cpp (hunk 3 above)
Comment (Ypostolov) [reply week_2/bus_stops_3/bus_stops_3.cpp:13]: We need a container set to facilitate the sorting of bus stops according to the terms of reference:
`наборы остановок, которые можно получить друг из друга перестановкой элементов или добавлением/удалением повторяющихся, следует считать одинаковыми.`

File: week_2/bus_stops_3/bus_stops_3.cpp (hunk 4 above)
Comment (Ypostolov) [reply week_2/bus_stops_3/bus_stops_3.cpp:23]: in: stop1 stop2 stop3
out: New bus 1 
route will be '1'

File: week_2/bus_stops_3/bus_stops_3.cpp (hunk 5 above)
Comment (Ypostolov) [reply week_2/bus_stops_3/bus_stops_3.cpp:30]: To have possibility to use this var for different routes.

**Review by ZhdanovP — APPROVED** [review]

**Pull request merged**